kuma/users/views.py

```diff
@@ -685,9 +685,23 @@ def get_form(self, form_class=None):
 
         # This magic sauce only exists whilst this Django view is being used
         # both by Yari (in testing) and by old Kuma (production).
-        # Once Kuma id divorced from doing any
+        # Once Kuma is divorced from doing any HTML rendering, we can refactor this.
         is_yari_signup = self.request.session.get("yari_signup", False)
         if is_yari_signup:
+
+            if not email:
+                # Pick the first primary email.
+                for data in extra_email_addresses:
+                    if data["primary"]:
+                        email = data["email"]
+                        break
+                else:
+                    # Pick the first non-primary email.
+                    for data in extra_email_addresses:
+                        email = data["email"]
```

Comment:
Or we could do something like this instead:
```suggestion
            if not email and extra_email_addresses:
                # Pick the first primary email.
                for data in extra_email_addresses:
                    if data["primary"]:
                        email = data["email"]
                        break
                else:
                    # Pick the first non-primary email.
                    email = extra_email_addresses[0]["email"]
```